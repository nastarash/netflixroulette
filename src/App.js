import React, { Fragment } from 'react';
import Header from './components/header/Header';
import Footer from './components/footer/Footer';
import SearchButton from './components/searchButton/SearchButton'

function App() {
  return (
    <Fragment>
      <Header />
      <SearchButton />
      <Footer />
    </Fragment>
  );
}

export default App;
