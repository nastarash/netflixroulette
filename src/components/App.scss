@import '~CommonStyles';

html,
body {
  margin: 0;
  padding: 0;
  height: 100%;


}

.page-container {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  background-color: $main-bg-color;

  &__content {
    padding: 30px 50px 50px 50px;
    flex-grow: 1;
    background-color: $light-bg-color;
  }
}
