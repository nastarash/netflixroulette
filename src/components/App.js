import React, { Component } from 'react';
import './App.scss';
import { AppErrorBoundary } from 'CommonComponents/AppErrorBoundary';
import { PageHeader } from 'CommonComponents/PageHeader';
import { PageFooter } from 'CommonComponents/PageFooter';
import { FilmResultsContainer } from 'CommonComponents/FilmResultsContainer';
import { Provider } from 'react-redux';
import { store, persistor } from 'Common/store';
import { PersistGate } from 'redux-persist/lib/integration/react';

export class App extends Component {
  render() {
    return (
      <AppErrorBoundary>
        <Provider store={store}>
          <PersistGate persistor={persistor}>
            <div className='page-container'>
              <PageHeader />
              <div className='page-container__content'>
                <FilmResultsContainer />
              </div>
              <PageFooter />
            </div>
          </PersistGate>
        </Provider>
      </AppErrorBoundary>
    );
  }
}


