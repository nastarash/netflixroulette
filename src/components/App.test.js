import React from 'react';
import { App } from './App';

describe('<App />', () => {
  test('should be rendered correctly', () => {
    const app = shallow(
      <App />
    );
    expect(app).toMatchSnapshot();
  });
});