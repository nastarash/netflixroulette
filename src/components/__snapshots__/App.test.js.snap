// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should be rendered correctly 1`] = `
<AppErrorBoundary>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <PersistGate
      loading={null}
      persistor={
        Object {
          "dispatch": [Function],
          "flush": [Function],
          "getState": [Function],
          "pause": [Function],
          "persist": [Function],
          "purge": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <div
        className="page-container"
      >
        <PageHeader />
        <div
          className="page-container__content"
        >
          <Connect(FilmResults) />
        </div>
        <PageFooter />
      </div>
    </PersistGate>
  </Provider>
</AppErrorBoundary>
`;
